File: src (cpp)/animal/reptilia/python.cpp
#include "python.h"

Python::Python(int _x, int _y, int _weight) : def_ID(9),
                                              def_ratio_meat(40),
                                              def_ratio_plant(0) {
  ID = def_ID;
  position.SetX(_x);
  position.SetY(_y);
  ratio_meat = def_ratio_meat;
  ratio_plant = def_ratio_plant;
  is_land_animal = true;
  is_water_animal = false;
  is_air_animal = false;
  weight = _weight;
  limb_count = 0;
}
void Python::Interact() {
  std::cout << "The python is slithering across the ground" << std::endl;
}

File: src/virtualzoo/animal/reptilia/Python.java
package virtualzoo.animal.reptilia;

import virtualzoo.animal.behavior.BehaviorTame;
import virtualzoo.animal.diet.Carnivora;

/**
 * Kelas Python mendefinisikan atribut untuk python.
 *
 * @author Rionaldi Chandraseta - 13515077
 * @version 3.0
 * @since 3.0
 */
public class Python extends Reptilia {

  /**
   * Constructor
   *
   * Menciptakan python dengan posisi (locX,locY) dan berat animalWeight.
   *
   * @param locX Nilai absis posisi.
   * @param locY Nilai ordinat posisi.
   * @param animalWeight Nilai berat python.
   */
  public Python(int locX, int locY, int animalWeight) {
    super();
    limb_count = 0;
    animalBehavior = new BehaviorTame();
    animalDiet = new Carnivora(animalWeight, 40);
    ID = 9;
    position.setLocation(locX, locY);
    isLandAnimal = true;
    isWaterAnimal = false;
    isAirAnimal = false;
  }

  /**
   * Mengatur interaksi dengan python.
   */
  public void interact() {
    System.out.println("The python is slithering across the ground");
  }
}
